src/aliceVision/calibration/checkerDetector.cpp

```diff
@@ -16,69 +22,74 @@ template <typename Type, int Size>
 using RowVector = Matrix<Type, 1, Size>;
 } // namespace Eigen
 
-namespace aliceVision{
-namespace calibration{
-
-
+namespace aliceVision {
+namespace calibration {
 
-bool CheckerDetector::process(const image::Image<image::RGBColor> & source)
+bool CheckerDetector::process(const image::Image<image::RGBColor>& source, bool useNestedGrids, bool debug)
 {
     image::Image<float> grayscale;
     image::ConvertPixelType(source, &grayscale);
 
+    const Vec2 center(grayscale.Width() / 2, grayscale.Height() / 2);
+
     const double scales[] = {1.0, 0.75, 0.5, 0.25};
 
-    std::vector<Vec2> allCorners;
+    std::vector<IntermediateCorner> allCorners;
     for (double scale : scales)
     {
+        ALICEVISION_LOG_INFO("[CheckerDetector] extracting corners at scale " << scale);
         std::vector<Vec2> corners;
         if (!processLevel(corners, grayscale, scale))
         {
+            ALICEVISION_LOG_DEBUG("[CheckerDetector] detection failed");
             return false;
         }
 
-        //Merge with previous level corners
+        ALICEVISION_LOG_DEBUG("[CheckerDetector] detected " << corners.size() << " corners");
+
+        // Merge with previous level corners
         for (Vec2 c : corners)
         {
-            double distmin = 5.0;
+            bool keep = true;
 
-            for (Vec2 oc : allCorners)
+            for (const IntermediateCorner & oc : allCorners)
             {
-                double dist = (oc - c).norm();
+                double dist = (oc.center - c).norm();
                 
-                if (dist < distmin)
+                if (dist < 5.0)
                 {
-                    distmin = dist;
+                    keep = false;
                     break;
                 }
             }
 
-            if (distmin < 5.0)
-            {
-                continue;
-            } 
+            if (!keep) continue;
 
-            allCorners.push_back(c);
+            IntermediateCorner sc;
+            sc.center = c;
+            sc.scale = scale;
+            allCorners.push_back(sc);
         }
     }
 
-    //Normalize image between 0 and 1
+    ALICEVISION_LOG_DEBUG("[CheckerDetector] kept " << allCorners.size() << " corners positions after merge between levels");
+
+    // Normalize image between 0 and 1
     image::Image<float> normalized;
     normalizeImage(normalized, grayscale);
 
-    // David Fleet and Tomas Pajdla and Bernt Schiele and Tinne Tuytelaars. ROCHADE: Robust Checkerboard Advanced Detection for Camera Calibration
-    std::vector<CheckerBoardCorner> fitted_corners;
-    fitCorners(fitted_corners, allCorners, normalized);
+    std::vector<CheckerBoardCorner> fittedCorners;
+    fitCorners(fittedCorners, allCorners, normalized);
 
-    //Remove multiple points at the same position
-    for (int i = 0; i < fitted_corners.size(); i++)
+    // Remove multiple points at the same position
+    for (int i = 0; i < fittedCorners.size(); i++)
```

Comment:
prefer preincrement and same type as `size()` for loops
```suggestion
    for (std::size_t i = 0; i < fittedCorners.size(); ++i)
```